### PR TITLE
plone.autoform.directives and more generic javascript

### PR DESCRIPTION
**Pull request by thet**
- Make use of more generic selectors in JavaScript, so that JavaScript works also for derived classes.
- Configure custom css classes for all event behavior fields.
  This makes it easier to use same selectors also for derived behaviors.
- Use ``plone.autoform.directives`` for manipulating field widgets instead of overriding the default Fieldwidget adapters.

### CODE REVIEW

File: plone/app/event/dx/behaviors.py [hunk 1]
```diff
@@ -44,10 +42,10 @@
 
 if PLONE5:
     # Plone 5
-    from plone.app.z3cform.widget import DatetimeWidget
+    from plone.app.z3cform.widget import DatetimeFieldWidget
 else:
     # Plone 4
-    from plone.app.widgets.dx import DatetimeWidget
+    from plone.app.widgets.dx import DatetimeFieldWidget
```
Comment (jensens) [inline plone/app/event/dx/behaviors.py]: if you skip Plone 4 support this can be removed as well?

Comment (thet) [inline plone/app/event/dx/behaviors.py]: True - But removing Plone 4 support is more than that. I'd like to make a separate PR for that.

**Comment by thet** [comment]
I'm wondering if there is a programmatic way to be used in an upgrade step to "re-cook" javascript...

**Comment by vincentfretin** [comment]
`portal_javascripts.cookResources()`

**Comment by thet** [comment]
@vincentfretin although it's from the plone-legacy bundle, but in Plone 5 it's different, no?

**Comment by vincentfretin** [comment]
Ah, I don't know about Plone 5.

File: plone/app/event/dx/behaviors.py (hunk 1 above)
Comment (jensens) [inline plone/app/event/dx/behaviors.py]: ok, one after another.

**Comment by jensens** [comment]
what’s the state here? do we need some "cooking" or is this sufficient?
